app/views/play/level/LevelGoals.vue

```diff
@@ -36,16 +36,32 @@
   LevelGoal = require('./LevelGoal').default
 
   module.exports = Vue.extend({
-    props: ['showStatus']
-    
-    data: -> {
-      overallStatus: ''
-      timedOut: false
-      goals: [] # TODO: Get goals, goalStates from vuex
-      goalStates: {}
-      casting: false
-    },
-    
+    props: {
+        showStatus:
+          type: Boolean
+          default: false
+
+        overallStatus:
```

Comment:
Is there any downside of moving all the state from `data` to `props`? From what I know, the vue component should not mutate props and use only the data for mutating values/state. In such a case, would it be a good idea if the Vue component wants to change the state?